src/driver-hidpp20.c

```diff
@@ -1155,9 +1178,11 @@ hidpp20drv_read_profile_8100(struct ratbag_profile *profile)
 		 * sensors is too niche to care about right now */
 		sensor = &drv_data->sensors[0];
 
-		ratbag_resolution_set_resolution(res,
-						 p->dpi[res->index],
-						 p->dpi[res->index]);
+		dpi = p->dpi[res->index];
+		ratbag_resolution_set_resolution(res, dpi, dpi);
+
+		if (dpi == 0)
+			res->is_disabled = true;
```

Comment:
If the dpi is disabled when reading the profile we should default to a sane value, at least for now.

```c
dpi = p->dpi[res->index];

if (dpi == 0)
	dpi = 800;

ratbag_resolution_set_resolution(res, dpi, dpi);
```

Reply:
I disagree here. One the side I don't thinks it's really necessary, one the other side I don't like the idea that we set an arbitrary default value. If I want resolutions to be disabled, let them be either zero or a value I've actually chosen, but not something "random". Doesn't add any value for the user, it rather confuses them where the 800 dpi is coming from.

Reply:
This is what we do when for ex. a device does not support reading the dpi values. 

Let's maybe try to look at this from a user perspective. Right now, if I disable a resolution and replug the device or restart the daemon, that resolution will show up as a disabled resolution with 0dpi. What I am proposing is that instead of having the resolution show up as 0dpi, have it show up as 800dpi. I totally agree that it is a bit strange, and it's not the best behavior but right now what can we do? I think it's better to have the resolution show up as 800dpi than 0dpi.

Reply:
As I said I don't like it (and I'm a user too). The `ratbagctl` is more a developer tool anyway, so I would like to keep it close to what's actually stored on the device, and in Piper this problem doesn't really exist, since you'll have the sliders anyway once you enable a resolution. I think in Logitech's software the behavior is quite similar, disabled resolution are greyed out and have 0 dpi.

Reply:
Sure, I understand. Maybe @bentiss could weight in.

Reply:
I'll just go with the minimum resolution then, that seems the most reasonable to me.